modules/apps/frontend-js/frontend-js-test/src/testFunctional/macros/CheckPeriodicExtendSessionRequests.macro

```diff
@@ -0,0 +1,15 @@
+definition {
+
+	macro getPeriodicExtendSessionRequests {
+		var time = "1";
+
+		while ("${time}" != "30") {
+			JSONSessionAutomationExtension.getSessionAutomationExtensionStatus();
+
+			var time = ${time} + 1;
+		}
+
+		echo("View periodic extend session requests successfuly");
+	}
```

Comment:
Maybe I'm not understanding the logic here but to me, it looks like this macro will always echo "View periodic extend session requests successfully" no matter what status is returned.

Also, `session.timeout.auto.extend.offset=115` means session is extended every 5 seconds but the while loop would occur almost immediately 30 times right after the other...how are you ensuring that the session requests are checked after 5 seconds?

Reply:
I will add a judge to set 'remember me = false' or 'extendSessionStatus = 200' it's can echo "View periodic extend session requests successfully"

Reply:
Maybe I still have a problem with how to confirm the regular session request in the network tab in the browser
I use curl -X GET -o '/dev/null' -w '%{http_code}' http://localhost:8080/c/portal/extend_session when I am not logged in, do I still use this when I am logged in, if Using the same, the returned value is 200 with no change

Reply:
I need to find a proper way to implement view periodic extend_session request, for this question I asked in the POSH group

Reply:
I'm not sure if we're going for the correct approach here. By running the GET curl we're actually getting the status ourselves, but we're not checking whether it periodically appears in the Network tab of dev Tools. I have run:
`
var curl = '''
				${baseURL}/c/portal/extend_session \
				-s \
				-D - \
			    -o /dev/null
			''';
`
and in response I got Headers of my GET, but nothing more. So, with this we're checking that the status is 200 for the whole time of requesting it, but we are not checking whether in dev tools it the `extend_session` appears periodically.

I believe we need to perform something similar to [this](https://stackoverflow.com/questions/45847035/using-selenium-how-to-get-network-request) example

Reply:
@john-co I think I might need some help, I might not be able to figure it out myself

Reply:
Sorry for the long delay getting back to you @Hijky. I did some google searches and there's some possible solutions but I'm not sure how feasible they are. Some ideas to explore:
https://stackoverflow.com/questions/45847035/using-selenium-how-to-get-network-request
https://stackoverflow.com/questions/67613901/how-to-fetch-status-of-javascript-and-css-requests-from-network-in-selenium-pyth

I'm wondering if just javascript can be used to listen to and retrieve network requests...maybe can reach out to FI devs in slack channel for tips on this.

Reply:
I find something about it in slack https://liferay.slack.com/archives/CD7939WBE/p1641922982010000